### PR TITLE
add use_if_null_to_convert_null_to_bool

### LINKED ISSUE #2460: new lint: `if_null_boolean_conversions` (use ?? to convert null to a boolean value.)
I'm sure someone can come up with a better name but the rationale is in https://github.com/dart-lang/linter/pull/2458#issuecomment-780088506.  The advice from effective Dart is 

**DO use ?? to convert null to a boolean value.**

See: https://dart.dev/guides/language/effective-dart/usage#do-use--to-convert-null-to-a-boolean-value

Cheers @a14n for the suggestion 🍻 .

### PR DESCRIPTION
**Pull request by a14n**
# Description

Use if-null operator to convert null to bool.

**BAD:**
```
if (nullableBool == true) {
}
```

**GOOD:**
```
if (nullableBool ?? false) {
}
```

Fixes #2460

### CODE REVIEW

**Review by bwilkerson — COMMENTED** [review]
Do we want to catch cases where the operands are reversed (`true == e`)?
Do we want to catch uses of `!=` (`e != false`)?

**Comment by a14n** [comment]
> Do we want to catch cases where the operands are reversed (`true == e`)?

I don't want to bother yoda conditions lovers :)  (this could be another lint BTW - `avoid_yoda_conditions`)

> Do we want to catch uses of `!=` (`e != false`)?

I will add some code to catch that.

**Review by bwilkerson — APPROVED** [review]

File: lib/src/rules/use_if_null_to_convert_null_to_bool.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Use if-null operator to convert null to bool.';
```
Comment (pq) [inline lib/src/rules/use_if_null_to_convert_null_to_bool.dart:13]: =>

'Use if-null operators to convert nulls to bools.';

(We standardized on plurals -- mostly 😬 )

File: lib/src/rules/use_if_null_to_convert_null_to_bool.dart [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+const _details = r'''
+
+Use if-null operator to convert null to bool.
```
Comment (pq) [inline lib/src/rules/use_if_null_to_convert_null_to_bool.dart:17]: =>

Use if-null operators to convert nulls to bools.

**Review by pq — COMMENTED** [review]
Consider renaming to `use_if_null_to_convert_nulls_to_bools`?

I feel like we've mostly pluralized.  For example:

`use_key_in_widget_constructors`
`use_late_for_private_fields_and_variables`
`use_setters_to_change_properties`

etc.

**Comment by pq** [comment]
PS: this is super awesome!

**Review by pq — APPROVED** [review]
Looks great!

**Comment by pq** [comment]
Thanks!

**Pull request merged**